### PR TITLE
Feature/auto deploy

### PR DESCRIPTION
**Pull request by damianavila**
This implement the things discussed in #1788 (large thread with go and forwards, read it all to get the complete picture).
Essentially, any time you want a devel build, you just need to tag and push the tag to origin an viola, after a :coffee: you have the builds available in Binstar.
How it is implemented? conda-build >> conda convert >> binstar upload, ALL of this happening in a TravisCI run. 

* [x] Make sure the builds are only generated by a tagged push
* [x] Use TRAVIS_BUILD_ID as a counter and a way to fingerprint the packages
* [x] Make the changes to also upload CDN and JS devel files to the CDN
* [x] Test all build process in a mirror repo, check here:
  * [TravisCI log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/49246085/log.txt)
  * [Packages available on Binstar](https://binstar.org/bokeh/bokeh/files?version=0.7.2.dev.49246084&page=1)
  * [JS and CSS files on the CDN](http://cdn.pydata.org/bokeh/dev/bokeh-0.7.2.dev.49246084.min.js)
* [x] Encrypted SHH to upload the devel docs.

As you see, I am almost done... today I had to fight with TravisCI builds not working in my test mirror, without any technical issue for not starting (I contact TravisCI support and they were surprises about the issue). Finally working now as I showed above.
I also hit a limitation in the log output that would be solved later with a new `--quiet` option in the conda convert command.
I have to make the SSH encryptation step and this would be ready to hit master after the proper review. I submitted here as WIP for early feedback. 

Thanks.

### CODE REVIEW

**Comment by bryevdv** [comment]
@damianavila should we go ahead and merge this and add the docs deploy in another PR?

**Comment by damianavila** [comment]
The docs deploys is almost there... I think it will be ready by night (maybe sooner), so let's wait until night, if I found some problem then we can split the PRs...

**Comment by damianavila** [comment]
OK, this is ready for review... it also fixes #1692.
There is still some things to do... but the main functionality is working as expected.
I did not add the deploy script because writing
```bash
git tag -a XXX-dev-xxxxxxxxx
git push origin XXX-dev-xxxxxxxxx
```
is simple enough to do it by hand, although we can add the `deploy` script later if needed.

**Comment by bryevdv** [comment]
The fact that it has duplicated information that must be carried correctly between both commands is enough reason to put it in a single script, IMO.

**Comment by damianavila** [comment]
OK, I will quickly add it...

**Comment by damianavila** [comment]
Done!

File: bokeh/resources.py [hunk 1]
```diff
@@ -42,11 +42,11 @@ def _get_cdn_urls(version=None, minified=True):
     base_url = _cdn_base_url()
     rel_container = 'bokeh/release'
     dev_container = 'bokeh/dev'
-    container = rel_container if dev != 'dev' else dev_container
+    container = rel_container if not dev.startswith(('dev', 'rc')) else dev_container
```
Comment (bryevdv) [inline bokeh/resources.py]: `container = dev_container if dev.startswith(('dev', 'rc')) else rel_container` would read easier to my eyes

Comment (damianavila) [inline bokeh/resources.py]: jaja... `-(-1) = 1` :stuck_out_tongue: but fair enough, I will change it...

Just as a comment, there is a reason I write it in that way... for western people, you read from left to right AND from top to bottom, so... as I first defined `rel_container` and then `dev_container` is is natural for me to make the construction with `rel_container` as the default condition... so, to be consistent with my idea, I will change this one to:
```python
     dev_container = 'bokeh/dev'
     rel_container = 'bokeh/release'
     container = dev_container if dev.startswith(('dev', 'rc')) else rel_container
```

Yes, I know this is subtle difference, but I firmly believe this kind of details are important from the readability point of view... OK, no more "nonsense linguistic", jaja...

**Comment by damianavila** [comment]
Comment addressed...

**Comment by bryevdv** [comment]
LGTM. A few comments for the future: It looks like conda `no-arch` is finally happening. I think we should move ahead with this, then when `no-arch` is released and available in conda widely, we can change from conda convert to building `no-arch` packages (on Travis) and then later, separately we can investigate moving the builds and some of the testing to binstar.org

**Comment by damianavila** [comment]
>LGTM. A few comments for the future: It looks like conda no-arch is finally happening. I think we should move ahead with this PR as-is,

I agree...

>then when no-arch is released and available in conda widely, we can change from conda convert to building no-arch packages (on Travis)

Yep

>and then later, separately we can investigate moving the builds and some of the testing to binstar.org

Yep... this is all in my plans :wink:

**Comment by damianavila** [comment]
I will left this open for a little bit, but I will merge it by today so we can upload a devel build tomorrow...

**Pull request merged**